mesonbuild/dependencies/base.py

```diff
@@ -2137,6 +2140,79 @@ def log_tried(self):
         return 'framework'
 
 
+class DependencyFactory:
+
+    """Factory to get dependencies from multiple sources.
+
+    This class provides a constructor that takes a set of names and classes
+    for various kinds of dependencies. This simplifies getting a single
+    dependency from multiple sources.
+
```

Comment:
(something like) "... calling the object returned by the constructor returns a list of constructors for Dependency-subclass objects to be tried in order" ?